apps/api/src/app/user-sessions/user-sessions.service.ts

```diff
@@ -141,9 +141,34 @@ export class UserSessionsService {
   }
 
   async remove(id: number, user: User) {
-    const userSession = await this.findOne(id, user);
+    const userSession = await this.findOneWithSessionData(id);
+
+    if (userSession.session.ownerId !== user.id && userSession.ownerId !== user.id) {
```

Comment:
Should this be an “or” condition?

Reply:
It is supposed to be that if you are not the owner of the session nor the owner of the user session, an error would be returned ... that enables the owner of the session to also remove user sessions which he typically doesn't own